ludwig/config_validation/checks.py

```diff
@@ -606,3 +607,38 @@ def check_qlora_requirements(config: "ModelConfig") -> None:  # noqa: F821
 
     if config.quantization and (not config.adapter or config.adapter.type != "lora"):
         raise ConfigValidationError("Fine-tuning and LLM with quantization requires using the 'lora' adapter")
+
+
+@register_config_check
+def check_prompt_task_and_template(config: "ModelConfig") -> None:  # noqa: F821
+    """Checks that prompt's template and task properties are valid, according to the description on the schema."""
+    if config.model_type != MODEL_LLM:
+        return
+
+    # If no prompt is provided, no validation necessary:
+    if not config.prompt:
+        return
+
+    template = config.prompt.template
+    task = config.prompt.task
+    # If no template is provided, task is required:
+    if template and not task:
+        raise ConfigValidationError("A prompt task is required if no template is provided!")
+
+    # If a template is provided, validate that it has a task and sample or some input column:
+    if template:
+        column_names = {feature.column for feature in config.input_features}
+        template_refs = set(Formatter().parse(template))
+        intersection = column_names & template_refs
+
+        if task and "__task__" not in template_refs:
+            raise ConfigValidationError(
+                "When providing a template and a task, you must make sure that the task keyword `{{__task__}} is "
+                "present somewhere in the template string!"
+            )
+
+        if len(list(intersection)) == 0 and "__sample__" not in template_refs:
```

Comment:
nit: `len(intersection)` should return the same value here (if I'm interpreting this right and `intersection` is a set)